### ISSUE TITLE
np.remainder gives strange result for integer zero

### ISSUE THREAD

**Issue by ahaldane** [numpy/numpy#6318]
I was looking through numpy ufuncs, and noticed `np.reciprocal` treats integer zero strangely:

    >>> np.reciprocal(np.array(0))
    -9223372036854775808

 Probably should return 0 with a warning, like `np.int32(1)/np.int32(0)`.  Thus for ints, reciprocal would return 1 if x==1, otherwise 0.

**Comment by pitrou** [comment]
Unfortunately there's no sane way to treat it. Integers have no `nan`, and ufuncs generally don't want to raise an exception for a single domain error amongst the input. `-9223372036854775808` is a bit special as it's the only int64 value whose opposite cannot be represented, as it's not entirely silly to use it as a "special value".
Returning 0 doesn't sound better to me...

**Comment by ahaldane** [comment]
I don't think it should raise an error, but a warning. Like the `np.divide` ufunc does in the equivalent case:

    >>> 1/np.arange(6)
    RuntimeWarning: divide by zero encountered in divide
    array([0, 1, 0, 0, 0, 0])

**Comment by eric-wieser** [comment]
Maybe related to #10374
